pkg/sbproxy/notifications/producer.go

```diff
@@ -159,39 +159,36 @@ func (p *Producer) run(ctx context.Context, messages chan *Message, group *sync.
 	p.lastNotificationRevision = unknownRevision
 	go p.scheduleResync(ctx, resyncChan, messages)
 	for {
-		needResync := p.lastNotificationRevision == unknownRevision
-		if err := p.connect(ctx); err != nil {
-			log.C(ctx).WithError(err).Error("could not connect websocket")
-			if err == errLastNotificationGone { // skip reconnect delay
-				p.lastNotificationRevision = unknownRevision
-				continue
-			}
-		} else {
-			if needResync {
-				log.C(ctx).Info("Last notification revision is gone. Triggering resync")
-				resyncChan <- struct{}{}
+		if err := p.connect(ctx, resyncChan); err != nil {
+			if err != errLastNotificationGone {
+				// Wait for the reconnect delay to pass before attempting to reconnect
+				log.C(ctx).WithError(err).Error("could not connect websocket")
+				select {
+				case <-ctx.Done():
+					log.C(ctx).Info("Context cancelled. Terminating notifications handler")
+					return
+				case <-time.After(p.producerSettings.ReconnectDelay):
+					log.C(ctx).Debug("Reconnection delay elapsed. Reattempting to establish websocket connection")
```

Comment:
"reestablish" is much more popular than "reattempt", see https://trends.google.com/trends/explore?geo=US&q=reattempt,reestablish